### PR TITLE
feat(GUI/all): added build system + dependencies, WindowManager, basic App

### PR DESCRIPTION
**Pull request by Sinan-Karakaya**
This PR does the following:

- added build system with dependency fetching
- added basic App class
- added WindowManager with ImGui integration
- added networkManager
- added basic init to test server
- added basic entities hierarchy (not implemented)

The code has been tested and no problems were found on my side.

### CODE REVIEW

File: zappy_gui/src/Utils/Utils.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+** src/Utils
+** File description:
+** Utils
+*/
+
+#include "Utils.hpp"
+
+const std::vector<std::string> &zp::Utils::split(const std::string &s, char delim)
+{
+    std::vector<std::string> *tokens = new std::vector<std::string>;
```
Comment (JasonLAGOUTE) [inline zappy_gui/src/Utils/Utils.cpp]: are you sure for the "new" ?

File: zappy_gui/src/WindowManager/WindowManager.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+{
+    ImGui::Begin("Chat");
+    ImGui::Text("This is the chat window");
+    ImGui::End();
+}
+
+void zp::WindowManager::drawGame()
+{
+    sf::Texture texture;
+    texture.loadFromFile("assets/sprites/background.png");
```
Comment (JasonLAGOUTE) [inline zappy_gui/src/WindowManager/WindowManager.cpp]: "assets/sprites/background.png" in a define

**Review by noerio — APPROVED** [review]
Code seems good, not documented for the moment

File: zappy_gui/src/WindowManager/WindowManager.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    ImGui::SFML::Render(m_window);
+    m_window.display();
+}
+
+void zp::WindowManager::setStyle()
+{
+    ImGuiIO& io = ImGui::GetIO(); (void)io;
+    io.IniFilename = "assets/imgui.ini";
+    io.Fonts->Clear();
+    io.Fonts->AddFontFromFileTTF("assets/fonts/Roboto-Medium.ttf", 16.0f);
```
Comment (JasonLAGOUTE) [inline zappy_gui/src/WindowManager/WindowManager.cpp]: "assets/fonts/Roboto-Medium.ttf" in a define

Comment (JasonLAGOUTE) [reply zappy_gui/src/WindowManager/WindowManager.cpp]: same for "16.0f"

**Review by JasonLAGOUTE — COMMENTED** [review]
Good code !

**Review by RoheeAxel — COMMENTED** [review]
Seems good

**Review by JasonLAGOUTE — COMMENTED** [review]
The code is good, for this PR it's fine but for the next one you will have to add documentation to avoid doing it at the end :)

File: zappy_gui/src/Entities/IEntity.hpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+#include <string>
+#include <array>
+
+#include <SFML/Graphics.hpp>
+
+namespace zp
+{
+
+enum Direction
```
Comment (thomasjuin1) [inline zappy_gui/src/Entities/IEntity.hpp:18]: Maybe do with cardinal position

**Review by thomasjuin1 — APPROVED** [review]
Code seems fine

**Review by noerio — APPROVED** [review]
Code seems good, not documented everywhere, wip

**Pull request merged**